pkg/route/api/validation/validation.go

```diff
@@ -276,35 +282,27 @@ func validateInsecureEdgeTerminationPolicy(tls *routeapi.TLSConfig, fldPath *fie
 
 // validateCertificatePEM checks if a certificate PEM is valid and
 // optionally verifies the certificate using the options.
-func validateCertificatePEM(certPEM string, options *x509.VerifyOptions) (*x509.Certificate, error) {
-	var data *pem.Block
-	for remaining := []byte(certPEM); len(remaining) > 0; {
-		block, rest := pem.Decode(remaining)
-		if block == nil {
-			return nil, fmt.Errorf("error decoding certificate data")
-		}
-		if block.Type == "CERTIFICATE" {
-			data = block
-			break
-		}
-		remaining = rest
+func validateCertificatePEM(certPEM string, options *x509.VerifyOptions) ([]*x509.Certificate, error) {
+	certs, err := cmdutil.CertificatesFromPEM([]byte(certPEM))
+	if err != nil {
+		return nil, err
 	}
 
-	if data == nil || len(data.Bytes) < 1 {
+	if len(certs) < 1 {
 		return nil, fmt.Errorf("invalid/empty certificate data")
 	}
 
-	cert, err := x509.ParseCertificate(data.Bytes)
-	if err != nil {
-		return nil, fmt.Errorf("error parsing certificate: %s", err.Error())
-	}
-
 	if options != nil {
-		_, err = cert.Verify(*options)
+		// Ensure we don't report errors for expired certs or if
+		// the validity is in the future.
+		options.CurrentTime = certs[0].NotAfter.AddDate(0, 0, -1)
+		_, err = certs[0].Verify(*options)
 		if err != nil {
-			return cert, fmt.Errorf("error verifying certificate: %s", err.Error())
+			if invalidErr, ok := err.(x509.CertificateInvalidError); !ok || invalidErr.Reason != x509.Expired {
```

Comment:
The original issue was to ensure cert+key+ca matched/was valid - which is why the extended validation was added. So ... we would need this.